radiance.go

```diff
@@ -108,56 +80,42 @@ func NewRadiance(opts client.Options) (*Radiance, error) {
 		return nil, fmt.Errorf("failed to create VPN client: %w", err)
 	}
 
-	// TODO: Ideally we would know the user locale to set the country on fronted startup.
-	f, err := newFronted(logWriter, reporting.PanicListener, filepath.Join(opts.DataDir, "fronted_cache.json"))
-	if err != nil {
-		log.Error("Failed to create fronted", "error", err)
-		return nil, fmt.Errorf("failed to create fronted: %w", err)
-	}
-	k := kindling.NewKindling(
-		kindling.WithPanicListener(reporting.PanicListener),
-		kindling.WithLogWriter(logWriter),
-		kindling.WithDomainFronting(f),
-		kindling.WithProxyless("api.iantem.io"),
-	)
-	// deviceid is a optional for macos, windows and linux
-	var platformDeviceId string
-	if common.IsAndroid() || common.IsIOS() {
-		platformDeviceId = opts.DeviceID
-	} else {
-		platformDeviceId = deviceid.Get()
-	}
-	// If no local setup from client options, use the default locale
-	if opts.Locale == "" {
-		log.Debug("Locale not set, using default en-US")
-		opts.Locale = "en-US"
-	}
-	userConfig := common.NewUserConfig(platformDeviceId, opts.DataDir, opts.DataDir)
-	u := api.NewUser(k.NewHTTPClient(), userConfig)
-	pro := api.NewPro(k.NewHTTPClient(), userConfig)
-	issueReporter, err := issue.NewIssueReporter(k.NewHTTPClient(), u, userConfig)
+	u := api.NewUser(init.kindling.NewHTTPClient(), init.userConfig)
+	issueReporter, err := issue.NewIssueReporter(init.kindling.NewHTTPClient(), u, init.userConfig)
 	if err != nil {
 		return nil, fmt.Errorf("failed to create issue reporter: %w", err)
 	}
-	confHandler := config.NewConfigHandler(configPollInterval, k.NewHTTPClient(), userConfig, opts.DataDir, vpnC.ParseConfig)
+	confHandler := config.NewConfigHandler(configPollInterval, init.kindling.NewHTTPClient(), init.userConfig, opts.DataDir, vpnC.ParseConfig)
 	confHandler.AddConfigListener(vpnC.OnNewConfig)
 
 	return &Radiance{
 		VPNClient:     vpnC,
 		confHandler:   confHandler,
 		activeServer:  new(atomic.Value),
 		stopChan:      make(chan struct{}),
-		user:          u,
-		pro:           pro,
-		userConfig:    userConfig,
+		userConfig:    init.userConfig,
 		issueReporter: issueReporter,
 		logsDir:       opts.LogDir,
 		shutdownFuncs: shutdownFuncs,
 	}, nil
 }
 
-// TODO: the server stuff should probably be moved to the VPNClient as well..
+// NewAPIHandler creates a new APIHandler instance. This is used to interact with the API.
+// The User and Pro fields are API clients used to communicate with the respective endpoints.
+//
+// Note: This separation is necessary because the iOS tunnel runs in a different isolated process.
+// and we need access to the API in the main process.
+func NewAPIHandler(opts client.Options) (*api.APIHandler, error) {
+	init, err := InitCommon(opts)
+	if err != nil {
```

Comment:
Since `User` and `Pro` are fields of the `APIHandler`, we don't really need to mention them here, unless they're function args or return values.

Adding the note is perfect!! We should just move it into the function since it's just to state the reason why this needs to be separate and that it shouldn't be changed.
```suggestion
// NewAPIHandler creates a new APIHandler instance. This is used to interact with the API.
func NewAPIHandler(opts client.Options) (*api.APIHandler, error) {
	// Note: This separation is necessary because the iOS tunnel runs in a different isolated process.
	// and we need access to the API in the main process.
	init, err := InitCommon(opts)
	if err != nil {
```

Reply:
Done.